tests/test_gpu_common.py

```diff
@@ -877,3 +879,39 @@ def test_npthreads(self):
         # Cannot provide a value larger than the thread pool size
         with pytest.raises(InvalidArgument):
             assert op.arguments(time_M=2, npthreads0=5)
+
+
+class TestEdgeCases(object):
+
+    def test_empty_arrays(self):
+        """
+        MFE for issue #1641.
+        """
+        grid = Grid(shape=(4, 4), extent=(3.0, 3.0))
+
+        f = TimeFunction(name='f', grid=grid, space_order=0)
+        f.data[:] = 1.
+        sf1 = SparseTimeFunction(name='sf1', grid=grid, npoint=0, nt=10)
+        sf2 = SparseTimeFunction(name='sf2', grid=grid, npoint=0, nt=10)
+        assert sf1.size == 0
+        assert sf2.size == 0
+
+        eqns = sf1.inject(field=f, expr=sf1 + sf2 + 1.)
```

Comment:
can you have something else than 1? Bit confusing whether it does nothing and stays at 1 or adds one here.

Reply:
does nothing -- see `npoint=0` above. sf1/sf2/sf -- they all are 0-size Functions

Reply:
I know that, but the equation is ambiguous IMO that's all, but GTG